pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluginwatcher
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"google.golang.org/grpc"
+)
+
+// PluginInfo contains gRPC connection info
+type PluginInfo struct {
+	SockPath string
+}
+
+// WatchPlugins watches for the creation of plugin sockets at the path
+func WatchPlugins(path string, outChan chan PluginInfo) error {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return err
+	}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+
+	err = watcher.Add(path)
+	if err != nil {
+		watcher.Close()
```

Comment:
what if `err == nil`?
Maybe things like `defer watcher.Close()`  should be addressed outside to ensure its collection